### PR TITLE
Add notification when fetch is skipped

### PR DESCRIPTION
**Pull request by azdrojowa123**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:103]: See @mkondratek's PR #837, `IntelliJNotificationCompat` will be gone

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:105]: Hardcoding `a minute` is questionable... note that we've got a constant for that, `com.virtuslab.gitmachete.frontend.actions.common.FetchUpToDateTimeoutStatus#FETCH_ALL_UP_TO_DATE_TIMEOUT_MILLIS`... maybe define another constant next to  `FETCH_ALL_UP_TO_DATE_TIMEOUT_MILLIS`, with a human-readable string like `"a minute"`

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:103]: You can actually base your PR on his

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:103]: `git machete github checkout-prs --all` FTW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: Pls name the param
```suggestion
        VcsNotifier.getInstance(project).notifyInfo(/* displayId */ null,
```

(at least IIRC that's `displayId`)

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: Note that this probably won't pass the binary compat check against 2020.1 ☹️

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: ...which is soon to be un-supported on #837

Comment (azdrojowa123) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: so wait for ... #837 ? because as I see all notifyInfo is available from 2020.3

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: Or even better, base your PR off @mkondratek 's branch

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: Which also implies waiting until #837 is merged to develop, but you get more relevant feedback from the CI

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java]: Hmm now the PR is based on #837 but the commits are still based on `develop`... pls actually rebase the commits on the top of `git-machete-tab-on-git-init`!

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java]: Generally, a branch layout like that:

image

File: frontend/base/src/main/resources/GitMacheteBundle.properties
Comment (PawelLipski) [inline frontend/base/src/main/resources/GitMacheteBundle.properties:67]: Btw you can mention this change in CHANGE_NOTES.html since it's user-visible

**Comment by PawelLipski** [comment]
You can actually rebase onto the most recent commit in base branch (bf7256d369a9b9de57932b84a75d0c49e8a2083c)... it seems to be based on some older commit of Mikołaj's PR.
In your IDE, in Git Machete tab: Fetch remotes + reset the diverged Mikołaj's branches into their remotes + rebase your branch.

**Comment by PawelLipski** [comment]
Gr8, perfect now :) lemme un-draft the PR, btw...

File: CHANGE-NOTES.html
Comment (PawelLipski) [inline CHANGE-NOTES.html:3]:
```suggestion
<p>The <i>Pull</i> action on specific branch displays notification about skipping fetch action when the last fetch was performed less than a minute ago.
```

**Comment by PawelLipski** [comment]
And I'll install the freshly-built plugin artifact from the CI for casual local tests (i.e. just to use in my regular work on other projects)... thanks to stacking the PRs, I'll test all 3 changes (2 of Mikołaj + this 1 of yours) together 🔱

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FetchUpToDateTimeoutStatus.java
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FetchUpToDateTimeoutStatus.java:9]: Nit: `TIME_UNIT` isn't the most fortunate name... imagine the timeout is 180 seconds... then the "time unit" would be `three minutes`... which isn't a time _unit_ as such (a `minute` or a `week` is, but not `three minutes`) 🤔 

I'd suggest just
```suggestion
  public static final String FETCH_ALL_UP_TO_DATE_TIMEOUT_AS_STRING = "a minute";
```

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchFastForwardOnlyAction.java:104]: image

we've got a little situation here Xd

Our pull (which is fast-forward merge underhood) sends a notification too (and it was first 🏅 !).
So what happens here is that within a minute after fetch...
1. pull action triggered
2. no fetch vcs notification send
3. fast forward merge notification send

and in result, no fetch notification shows only for a very short time (one cannot read it), which looks bad. 

we can:
1. show the no fetch notification somewhere else (not with vcs notifier)
2. merge notification messages in case we need both of them. smth like: `No fetch blah blah. Fast forwarded blah blah` 

In the case of the first option, we could use a standard notification with a linked action that would enforce a fetch - seems nice to me.

File: CHANGE-NOTES.html
Comment (mkondratek) [inline CHANGE-NOTES.html:4]: The changes to code refer fast forward merge - not the pull. 🤔 Probably it's applicable too but for now inconsistent.

**Review by mkondratek — CHANGES_REQUESTED** [review]

File: CHANGE-NOTES.html
Comment (PawelLipski) [reply CHANGE-NOTES.html:4]: Let's "stash away" this PR for a moment and revisit after/apply on the top of changes needed for issue #852

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:34]: Two-letter [acronyms & initialisms](https://www.rd.com/article/acronym-vs-abbreviation-whats-the-difference/) are fully capitalized in camel case as per Java conventions:
```suggestion
    val successFFMergeNotification = fetchNotification
```

Only up from three letters is upper case used (`Tls` and not `TLS`)

**Review by PawelLipski — APPROVED** [review]
A very neat & clear change now after the refactor has been performed separately <3

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java
Comment (azdrojowa123) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: Screenshot 2022-05-25 at 08 58 33
Screenshot 2022-05-25 at 08 57 24

Comment (azdrojowa123) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: Screenshot 2022
<img width=
-05-25 at 08 57 10" src="https://user-images.githubusercontent.com/33667003/170200532-1cc4c36d-a305-4947-b349-054edbce9890.png">

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: Gr8 🎉

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: hm. can we be consistent on `.` usage here? so add one at the end pls

**Review by mkondratek — APPROVED** [review]
pls look at the last minor comments. approving! 🚀

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java
Comment (azdrojowa123) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: or maybe better without `. ` ?

Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java:48]: Yeah, I'd say the **last** sentence in the notification should have **no** trailing `.`

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**
